tools/querytee/proxy.go

```diff
@@ -151,13 +152,31 @@ func NewProxy(cfg ProxyConfig, logger log.Logger, routes []Route, registerer pro
 }
 
 func (p *Proxy) Start() error {
-	// Setup listener first, so we can fail early if the port is in use.
-	listener, err := net.Listen("tcp", fmt.Sprintf(":%d", p.cfg.ServerServicePort))
+	// Setup server first, so we can fail early if the ports are in use.
+	serv, err := server.New(server.Config{
+		// HTTP configs
+		HTTPListenPort:                p.cfg.ServerServicePort,
```

Comment:
I would rename the config field for clarity.

```suggestion
		HTTPListenPort:                p.cfg.ServerHTTPServicePort,
```